tap_zendesk/streams.py

```diff
@@ -139,88 +141,110 @@ def sync(self, state):
 class Tickets(Stream):
     name = "tickets"
     replication_method = "INCREMENTAL"
-    replication_key = "updated_at"
+    replication_key = "generated_timestamp"
 
     last_record_emit = {}
-    def _buffer_record(self, buf, record):
+    buf = {}
+    buf_time = 60
+    def _buffer_record(self, record):
         stream_name = record[0].tap_stream_id
         if self.last_record_emit.get(stream_name) is None:
             self.last_record_emit[stream_name] = utils.now()
-        buf.append(record)
-        if (utils.now() - self.last_record_emit[stream_name]).total_seconds() > 300:
+
+        if self.buf.get(stream_name) is None:
+            self.buf[stream_name] = []
+        self.buf[stream_name].append(record)
+
+        if (utils.now() - self.last_record_emit[stream_name]).total_seconds() > self.buf_time:
             self.last_record_emit[stream_name] = utils.now()
             return True
+
         return False
 
+    def _empty_buffer(self):
+        for stream_name, stream_buf in self.buf.items():
+            for rec in stream_buf:
+                yield rec
+            self.buf[stream_name] = []
+
     def sync(self, state):
         bookmark = self.get_bookmark(state)
         tickets = self.client.tickets.incremental(start_time=bookmark)
         audits_stream = TicketAudits(self.client)
-        ticket_buffer = []
-        audit_buffer = []
+        metrics_stream = TicketMetrics(self.client)
 
-        def emit_audits_metric():
-            singer.metrics.log(LOGGER, Point(metric_type='counter',
-                                             metric=singer.metrics.Metric.record_count,
-                                             value=audits_stream.count,
-                                             tags={'endpoint':audits_stream.stream.tap_stream_id}))
-            audits_stream.count = 0
+        def emit_sub_stream_metrics(sub_stream):
+            if sub_stream.is_selected():
+                singer.metrics.log(LOGGER, Point(metric_type='counter',
+                                                 metric=singer.metrics.Metric.record_count,
+                                                 value=sub_stream.count,
+                                                 tags={'endpoint':sub_stream.stream.tap_stream_id}))
+                sub_stream.count = 0
 
         if audits_stream.is_selected():
             LOGGER.info("Syncing ticket_audits per ticket...")
 
         for ticket in tickets:
-            if utils.strptime_with_tz(ticket.updated_at) < bookmark:
+            generated_timestamp_dt = datetime.datetime.utcfromtimestamp(ticket.generated_timestamp).replace(tzinfo=pytz.UTC)
+            if generated_timestamp_dt < bookmark:
```

Comment:
You know, I wonder if this won't get hit anymore. I bet we were seeing that with `updated_at` for the same reason we're using `generated_timestamp` now.